BrainPortal/app/controllers/neurohub_application_controller.rb

```diff
@@ -83,6 +84,35 @@ def check_if_locked
 
   end
 
+  ########################################################################
+  # Messaging System Filters (presently only invite acceptance)
+  ########################################################################
+
+  # Find new messages to be displayed at the top of the page.
+  def prepare_messages #:nodoc:
+    return unless current_user
+    return if     current_user.all_licenses_signed.blank?
+    return if     request.format.blank? || request.xhr?
+    return unless request.format.to_sym == :html
+
+    @display_messages = []
+
+    unread_messages = current_user.messages.where( :read => false, :header => 'Invitation Accepted' ).order( "last_sent DESC" )
```

Comment:
This query is brittle by the way. If someone in the future change the header of the message that is being sent (see your code in InvitationControlelr and NhInvitationController), then this piece of code will stop displaying the message.

Reply:
Should I then add `message_type=confirmation` flag along with relevant styles? 

Showing all the cbrain messages seems a bit risky. Or as a half measure may be add a unittest?

Reply:
Just use the same query but encspsulate it in a method, like I explained above, so that you only have one 'prepare_message' callback in ApplicationController.